### PR TITLE
[CPDTP-565] - correct NPQ withdrawal code

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDTP-565

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (docs/source/release-notes)

### CODE REVIEW

File: spec/services/participants/withdraw/npq_spec.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@
       cpd_lead_provider: cpd_lead_provider,
       participant_id: npq_profile.user.id,
       course_identifier: "npq-leading-teaching",
-      reason: "left-teaching-profession",
+      reason: Participants::Withdraw::NPQ.reasons,
```
Comment (kicferk1) [inline spec/services/participants/withdraw/npq_spec.rb:15]: Does this need to be a sample?

Comment (StupidCodeFactory) [reply spec/services/participants/withdraw/npq_spec.rb:15]: that's very interesting that this test then 😲 ! Will look into that

File: spec/support/shared_examples/participant_actions_withdraw_support.rb [hunk 2]
```diff
@@ -49,4 +49,15 @@
...
+  context "with an invalid request" do
+    before { params[:data][:attributes][:reason] = "erroneous-reason" }
+
+    it "changes the training status of a participant to withdrawn" do
+      put url, params: params
+
+      expect(response).not_to be_successful
+      expect(parsed_response.dig("errors", 0, "detail")).to eq("Reason is not included in the list")
+    end
+  end
```
Comment (kicferk1) [inline spec/support/shared_examples/participant_actions_withdraw_support.rb:62]: "changes the training status of a participant to withdrawn"? Doesn't sound like what the test is doing

File: app/services/participants/withdraw/validate_and_change_state.rb [hunk 3]
```diff
@@ -9,8 +9,7 @@ module ValidateAndChangeState
       included do
         attr_accessor :reason
 
-        validates :reason, presence: true
-        validates :reason, inclusion: { in: reasons }, allow_blank: true
+        validates :reason, inclusion: { in: reasons }
```
Comment (kicferk1) [inline app/services/participants/withdraw/validate_and_change_state.rb:12]: Why are you removing the explicit presence check? The error messages should be different, I think.

File: app/services/participants/base.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@ def call(params:)
 
     def call
       unless valid?
-        raise ActionController::ParameterMissing, errors.map(&:full_message)
+        raise ActionController::ParameterMissing, errors.map(&:message)
```
Comment (StupidCodeFactory) [inline app/services/participants/base.rb:20]: this actually was a mistake from my part in the previous PR https://github.com/DFE-Digital/early-careers-framework/pull/1188

it actually changed the formatting of the error message.

File: app/services/participants/withdraw/npq.rb [hunk 5]
```diff
@@ -6,8 +6,20 @@ class NPQ < Base
...
+            insufficient-capacity
+            change-in-developmental-or-personal-priorities
+            change-in-school-circumstances
+            change-in-school-leadership
+            quality-of-programme-structure-not-suitable.
+            quality-of-programme-content-not-suitable
+            quality-of-programme-facilitation-not-effective
+            quality-of-programme-accessibility
+            quality-of-programme-other
+            programme-not-appropriate-for-role-and-CPD-needs
```
Comment (jamesrose) [inline app/services/participants/withdraw/npq.rb:22]:
```suggestion
            programme-not-appropriate-for-role-and-cpd-needs
```

File: swagger/v1/component_schemas/NPQParticipantWithdraw.yml [hunk 6]
```diff
@@ -5,10 +5,22 @@ properties:
...
+      - insufficient-capacity
+      - change-in-developmental-or-personal-priorities
+      - change-in-school-circumstances
+      - change-in-school-leadership
+      - quality-of-programme-structure-not-suitable.
+      - quality-of-programme-content-not-suitable
+      - quality-of-programme-facilitation-not-effective
+      - quality-of-programme-accessibility
+      - quality-of-programme-other
+      - programme-not-appropriate-for-role-and-CPD-needs
```
Comment (jamesrose) [inline swagger/v1/component_schemas/NPQParticipantWithdraw.yml:21]:
```suggestion
      - programme-not-appropriate-for-role-and-cpd-needs
```

**Pull request merged**